lib/Dialect/Verif/Transforms/PrepareForFormal.cpp

```diff
@@ -0,0 +1,69 @@
+//===- FlattenWires.cpp - Flattens all wires ------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//===----------------------------------------------------------------------===//
+//
+// Replace all wires with their inputs.
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "circt/Dialect/HW/HWOps.h"
+#include "circt/Dialect/HW/HWTypes.h"
+#include "circt/Dialect/Verif/VerifDialect.h"
+#include "circt/Dialect/Verif/VerifOps.h"
+#include "circt/Dialect/Verif/VerifPasses.h"
+#include "circt/Support/Namespace.h"
+#include "circt/Support/Naming.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+#include "mlir/Pass/Pass.h"
+
+using namespace circt;
+using namespace mlir;
+using namespace verif;
+
+namespace {
+/// Flatten wires by unconditionally replacing them with their inputs
+/// Wires create an alias for a set of operations, they are usually removed
+/// through canonicalization at some point.Some wires are however
+/// maintained.This pass unconditionally replaces all wires with their inputs,
+/// making it easier to reason about in contexts where wires don't exist.
+struct WireOpConversionPattern : OpConversionPattern<hw::WireOp> {
+  using OpConversionPattern<hw::WireOp>::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(hw::WireOp wire, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    rewriter.replaceOp(wire, adaptor.getInput());
```

Comment:
I'm not blocking but note that a wire could have an inner symbol. Verifier after the pass will fail if there is inner symbol user of it so it will be necessary to remove them as well in this pass.